### PR TITLE
fix: 'ipfs routing findpeer' explicitly fails when searching for self

### LINKED ISSUE #9898: Make clear that finding your own node in the DHT is currently not supported
## Done criteria
Let's not be silent when one does:

`ipfs dht findpeer {{YourOwnPeerID}}`

but instead return an error

`Finding your own node in the DHT is currently not supported`

## Notes
This came out of a discussion here: https://filecoinproject.slack.com/archives/C04M8232QRW/p1685007519645339

### PR DESCRIPTION
**Pull request by susarlanikhilesh**
Closes #9898

### CODE REVIEW

**Comment by BigLep** [comment]
@hacdias : can you provide a quick review please?

**Review by hacdias — COMMENTED** [review]
Seems fine, but perhaps a test would be great.

**Comment by susarlanikhilesh** [comment]
Not sure on how to write a test case for that scenario itself as we need the daemon running while we want to test this command. Any leads would be great. 

Thank you

**Comment by hacdias** [comment]
@susarlanikhilesh you can write a test inside `test/cli`. Those tests all use a daemon. Perhaps this file would be the best to insert a `t.Run("ipfs routing findpeer fails for self", func(t *testing.T) { })` or something like that.

**Comment by susarlanikhilesh** [comment]
@hacdias Can you please check if that's sufficient.

File: core/commands/routing.go [hunk 1]
```diff
@@ -301,6 +301,10 @@ var findPeerRoutingCmd = &cmds.Command{
 			return err
 		}
 
+		if pid == nd.Identity {
+			return ErrSelfUnsupported
```
Comment (Jorropo) [inline core/commands/routing.go:305]: I guess this works but this seems very ad-hoc.
I think this is covering up a bug, ideally the routing layers like dht, ... would return this error.

This is better than nothing still.

**Pull request merged**